paasta_tools/kubernetes_tools.py

```diff
@@ -2049,3 +2099,25 @@ def get_pod_hostname(kube_client: KubeClient, pod: V1Pod) -> str:
         return pod.spec.node_name
     # if label has disappeared (say we changed it), default to node name
     return node.metadata.labels.get("yelp.com/hostname", pod.spec.node_name)
+
+
+def to_node_label(label: str) -> str:
+    """k8s-ifies certain special node labels"""
+    if label in {"instance_type", "instance-type"}:
+        # kube_client is assumed to already be configured
+        version_info = kube_client.VersionApi().get_code()
```

Comment:
Where did `kube_client` come from? 
It looks like this line of code actually makes an api call and stuff. 

It is really too bad this function needs to do that?

Reply:
It's an alias: `from kubernetes import client as kube_client`. We use it in the `KubeClient` class, instances of which we also confusingly name `kube_client`.

An API call is indeed unfortunate but since the instance_type label differs by version, I can't really see a better choice to provide some level of compatibility protection. We could just drop this and make all services specify the full label, but then we'd have to change them if we ever upgrade. We could also preemptively upgrade k8s to v1.17, but the Python k8s client only supports up to v1.15.

Reply:
Adjusted, and removed API call to k8s.